### DISCUSSION TITLE
does binary include postgresql? or not?

### DISCUSSION THREAD

**Discussion by dataders** [psycopg/psycopg2#1621] [Q&A] [answered]
I've been told that the psycopg2-binary module does not depend on an underlying installed version of  `libpq`.

Does this mean that `libpq` itself is shipped with psycopg2 when installing the binary module? If so, is there a specific version of libpq that it comes with? Or is something else happening that I'm missing here?

The impetus for this thread is compatibility with shiny new postgres offerrings like Tembo & Neon. Both services' docs have the troubleshooting guides below that state that psql clients must be of version 14 or greater so that the Server Name Indication (SNI) can be done in TLS.

- [Tembo: Troubleshooting](https://tembo.io/docs/tembo-cloud/troubleshooting/#tembo-requires-postgresql-clients-of-version-14)
- [Neon: Connection Errors: The endpoint ID is not specified](https://neon.tech/docs/connect/connection-errors#the-endpoint-id-is-not-specified)

On my local machine (arm64/M1 MacOS), I can use psycopg2 to connect without specifying an `endpoint` because my psql client is version 15.

**Comment by dvarrazzo** [comment, marked as answer]
Psycopg2-binary comes with a libpq shipped, yes. The version shipped, for recent psycopg version, is > 14. The libpq shipped is usually declared in the [news file](https://www.psycopg.org/docs/news.html) and can be retrieved using the [`libpq_version()` function](https://www.psycopg.org/docs/extensions.html#psycopg2.extensions.libpq_version).
